app/test/e2e/endpoint.spec.js

```diff
@@ -0,0 +1,70 @@
+const nock = require('nock');
+const { TOKENS, microserviceTest } = require('./test.constants');
+const { initHelpers } = require('./utils');
+const { getTestAgent, closeTestAgent } = require('./test-server');
+const Microservice = require('models/microservice.model');
+const Endpoint = require('models/endpoint.model');
+
+const helpers = initHelpers(getTestAgent);
+
+let requester;
+
+const getListEndpoints = () => requester
+    .get('/api/v1/endpoint')
+    .set('Authorization', `Bearer ${TOKENS.ADMIN}`)
+    .send();
+
+const createMicroservice = () => requester
+    .post('/api/v1/microservice')
+    .set('Authorization', `Bearer ${TOKENS.ADMIN}`)
+    .send(microserviceTest);
+
+const openEnpoint = endpoint => requester
+    .get(endpoint)
+    .send();
+
+describe('Endpoints calls', () => {
+    before(async () => {
+        if (process.env.NODE_ENV !== 'test') {
+            throw Error(`Running the test suite with NODE_ENV ${process.env.NODE_ENV} may result in permanent data loss. Please use NODE_ENV=test.`);
+        }
+
+        requester = await getTestAgent();
+        helpers.setRequester(requester);
+        nock.cleanAll();
+    });
+
+    it('Getting a list of endpoints without being authenticated should fail', helpers.isTokenRequired('get', 'plugin'));
```

Comment:
please replace these with the actual code, as you see on the other tests. I understand the value of DRY-ing code, but it also makes the test less readable